File: cppTET/SearchDeepQ/DeepQ/test/NIPSBenchmark1.0.3/Agent/NIPSContinuousRandomAgent.java
import java.util.*;

public class NIPSContinuousRandomAgent {

Random random = new Random(666);
int minAction;
int actionRange;
Java_agent_parser jap = new Java_agent_parser();

public void agent_init(String ts)
{
    jap.parser(ts);
    minAction = (int)jap.action_mins[0];
    actionRange = (int)jap.action_maxs[0]-minAction;

}            

public int agent_start(float[] obs)
{
    return randomAction();
}

public void agent_cleanup()
{
}

public int agent_step(float r, float[] obs)
{
    return randomAction();
}

public void agent_end(float r)
{
}    

public String agent_get_name()
{
    return "tedmunds_Rutgers_NIPSContinuousRandomAgentJava";
}

private int randomAction()
{
    int value = random.nextInt(actionRange+1);
    return minAction+value;
}

}

File: cppTET/SearchDeepQ/DeepQ/test/NIPSBenchmark1.0.3/Agent/NIPSContinuousRandomAgent.cpp
#include "RandomAgent.h"
#include "RL_agent_util.h"
#include <cstdlib>
#include <cmath>
#include <cstdio>

using namespace std;

Action randomAction();

void agent_init(Task_specification ts)
{
	parse_task_spec(ts, &taskSpec);
	
	srand48(666);
}

Action agent_start(Observation o)
{
    return randomAction();
}

Action agent_step(Reward r, Observation o)
{
    return randomAction();
}

void agent_end(Reward r)
{
	//Dont care
}

void agent_cleanup()
{
	// Nothing to clean up
}

char* agent_get_name()
{
	return "tedmunds_Rutgers_NIPSContinuousRandomAgent";
}

Action randomAction()
{
	// Do the discrete dimensions
	int lookup = 0;
	// TODO: don't assume that every dimension has a max and a min
	int range = (int)floor(taskSpec.action_maxs[lookup] - taskSpec.action_mins[lookup]) + 1;
	long int randVal = lrand48();
	int value = randVal%range + (int)taskSpec.action_mins[lookup];
	action = value;
	return action;
}

